crates/bevy_ecs/src/world/command_queue.rs

```diff
@@ -120,7 +120,25 @@ impl CommandQueue {
         // flush the previously queued entities
         world.flush_entities();
 
-        self.apply_or_drop_queued(Some(world));
+        // Append our commands to the swap queue
+        world.swap_queue.append(self);
```

Comment:
Can we avoid this call to `append`? It will cause unnecessary allocations

I imagine that in most bevy apps, the world's command queue will be quite small, since it only stores commands enqueued by observers or hooks, whereas the size of system command queues will vary a lot. If there's a single system queues up a bunch of commands, this will force the size of the world's command queue to match that even when it's unnecessary

Reply:
Fair point, fixed.